yourgov/components/FormMobileFoodVendorPermit/task/FormStep7.tsx

```diff
@@ -1,72 +1,210 @@
-import { ComboboxMulti } from '@ag.ds-next/react/combobox';
+import { type FormEventHandler, useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import { Button, ButtonGroup, ButtonLink } from '@ag.ds-next/react/button';
+import { H2, H3 } from '@ag.ds-next/react/heading';
+import { AvatarIcon, PlusIcon } from '@ag.ds-next/react/icon';
+import { Modal } from '@ag.ds-next/react/modal';
+import { PageAlert } from '@ag.ds-next/react/page-alert';
 import { Stack } from '@ag.ds-next/react/stack';
-import { ShallowErrors } from '../FormState';
+import {
+	Table,
+	TableBody,
+	TableCell,
+	TableHead,
+	TableHeader,
+	TableRow,
+	TableWrapper,
+} from '@ag.ds-next/react/table';
+import { Text } from '@ag.ds-next/react/text';
+import { SectionAlert } from '@ag.ds-next/react/section-alert';
 import { StepActions } from '../StepActions';
 import { useGlobalForm } from '../GlobalFormProvider';
 import { FormContainer } from './FormContainer';
-import { useFormContext } from './FormProvider';
-import { step7FormSchema, type Step7FormSchema } from './FormState';
+import { taskFormSteps, useFormContext } from './FormProvider';
+import { Step7FormSchema } from './FormState';
 
 export function FormStep7() {
-	const { formState, step7SetState, isSavingBeforeExiting } = useGlobalForm();
+	const { step7GetState, step7SetState, isSavingBeforeExiting } =
+		useGlobalForm();
 	const { submitStep } = useFormContext();
+	const { query } = useRouter();
+	const step7State = step7GetState();
 
-	const {
-		control,
-		handleSubmit,
-		formState: { errors },
-	} = useForm<Step7FormSchema>({
-		defaultValues: formState.task?.step7,
-		resolver: isSavingBeforeExiting ? undefined : zodResolver(step7FormSchema),
-		mode: 'onSubmit',
-		reValidateMode: 'onBlur',
-	});
+	const [employeeToRemove, setEmployeeToRemove] =
+		useState<Partial<Step7FormSchema['employee']>>();
+	const [removedEmployee, setRemovedEmployee] = useState('');
 
-	const typeCorrectedErrors = errors as ShallowErrors<Step7FormSchema>;
+	const [modalIsVisible, setModalIsVisible] = useState(false);
+	const [showRemovedEmployeeMessage, setShowRemovedEmployeeMessage] =
+		useState(false);
+	const [showErrorMessage, setShowErrorMessage] = useState(false);
+
+	const addedEmployee = query.success
+		? step7State?.employee &&
+		  step7State.employee.find((employee) => employee?.id === query.success)
+		: undefined;
+	const [showAddedEmployeeMessage, setShowAddedEmployeeMessage] = useState(
+		!!addedEmployee
+	);
+	useEffect(() => {
+		setShowAddedEmployeeMessage(!!addedEmployee);
+	}, [addedEmployee]);
+
+	const closeModal = () => {
+		setShowAddedEmployeeMessage(false);
+		setEmployeeToRemove(undefined);
+	};
+
+	useEffect(() => {
+		setModalIsVisible(!!employeeToRemove);
+		setShowRemovedEmployeeMessage(false);
+	}, [employeeToRemove]);
+
+	const removeEmployee = () => {
+		step7SetState({
+			...step7GetState,
+			completed: false,
```

Comment:
If you remove an employee that isn't selected, should this remain `completed: true`?

Reply:
I don't think so. Because any action to this page still requires saving